compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -1188,4 +1236,46 @@ private boolean isWithinParenthesis(Token openParen, Token closeParen) {
         return !isParenthesisMissing(openParen, closeParen) &&
                 isPosWithinOpenCloseLineRanges(linePosition, openParen.lineRange(), closeParen.lineRange());
     }
+
+    private Optional<TypeSymbol> transformFunctionOrMethod(BLangNode bLangNode, BType variableType,
+                                                           BInvokableSymbol bLangOriginalLangLib,
+                                                           SeparatedNodeList<FunctionArgumentNode> arguments,
+                                                           Token openParenToken, Token closeParenToken,
+                                                           boolean hasFirstArg) {
+        if (bLangOriginalLangLib == null || bLangOriginalLangLib.params.size() == 0) {
+            return getExpectedTypeFromFunction(bLangNode, openParenToken, closeParenToken);
+        }
+
+        BVarSymbol firstParam = bLangOriginalLangLib.params.get(0);
+        BType typeParam = new TypeParamFinder().find(firstParam.getType());
+
+        if (typeParam == null) {
+            return getExpectedTypeFromFunction(bLangNode, openParenToken, closeParenToken);
+        }
+
+        BInvokableSymbol langLibMethod =
+                this.langLibFunctionBinder.cloneAndBind(bLangOriginalLangLib, variableType,
+                        SymbolUtils.getTypeParamBoundType(variableType));
+
+        if (langLibMethod.getParameters().size() < 2) {
```

Comment:
Shall we put a comment specifying why  `size >=2 ` condition should be satisfied (Just add a sample langlib method with parameters in the comment)